lxd/network/driver_bridge.go

```diff
@@ -2543,6 +2543,29 @@ func (n *bridge) getExternalSubnetInUse() ([]externalSubnetUsage, error) {
 	externalSubnets = append(externalSubnets, bridgeNetworkExternalSubnets...)
 	externalSubnets = append(externalSubnets, bridgedNICExternalRoutes...)
 
+	// Detect if there are any conflicting proxy devices on all instances with the to be created network forward
+	err = n.state.DB.Cluster.InstanceList(context.TODO(), func(inst db.InstanceArgs, p api.Project) error {
+		devices := db.ExpandInstanceDevices(inst.Devices, inst.Profiles)
+
+		for devName, devConfig := range devices {
+			if devConfig["type"] != "proxy" {
+				continue
+			}
+
+			proxyListenAddr, err := ProxyParseAddr(devConfig["listen"])
+			if err != nil {
+				return err
+			}
+
+			for netFwdAddr := range projectNetworksForwardsOnUplink {
+				if net.ParseIP(proxyListenAddr.Address).Equal(net.ParseIP(netFwdAddr)) {
+					return fmt.Errorf("A proxy device (%q) has an overlapping listen IP address with the network forward IP address (%q) you are creating", devName, netFwdAddr)
```

Comment:
We cant always be sure that this function is called in a creation context (it is at the moment, but could be changed in the future), so lets lose that part.

Also the proxy dev name is somewhat meaningless without the project and instance name too.

How about something like:

"Listen address %q conflicts with proxy device %q(devName) on %(instanceName) in project %q(projectName)"